src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -667,6 +697,9 @@ private static boolean isLoopAst(int ast) {
         /** Contains definitions of uninitialized variables. */
         private final Deque<DetailAST> uninitializedVariables = new ArrayDeque<>();
 
+        /** Whether there is a {@code break} or {@code return} in the scope. */
+        private boolean containsBreakOrReturn;
```

Comment:
Some more information.
We cannot directly assign a final variable in a loop at all.
```
final int a;
while (...) {
  a = 1; // compile error
}
```
We would get a `variable 'a' might be assigned in loop` error.
We can only assign a final variable (declared outside the loop) in the loop like this:
```
final int a;
while (...) {
  if (...) {
    a = 1;
    break; // necessary, or there would still be a compile error
  }
}
```
or this, weird but without compile error:
```
final int a;
while (...) {
  a = 1;
  break; // necessary
}
```
But if we try to access `a` outside the loop, we would get a `variable 'a' might not have been initialized` error.
So `a` must be assigned outside the loop if we want to access it. Then `a` should not be declared as final.
I assume the only case we should raise a violation is smth like this:
```
int a; // violation
while (...) {
  if (...) {
    a = 1;
    System.out.print(a);
    break;
  }
}
// never assign or access 'a' again
```
And the check won't raise that violation by now. In fact we have a judgement here https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java#L511-L513
> if the variable is declared outside the loop and initialized inside
> the loop, then it cannot be declared final, as it can be initialized
> more than once in this case

Therefor, whether I handle `break` in loop or not, no violation would be raised due to that judgement.

Is it really necessary to handle such weird cases like that? I believe no one would write code in that way.